snake.lua

```diff
@@ -113,6 +114,14 @@ snake.die = function(this)
         table.removeFromTable(GRID[v.x][v.y], this)
     end
 
+    for x,_v in ipairs(GRID) do
```

Comment:
This isn't very efficient and doesn't tackle the root issue of not deleting references to the dead snake in the grid, but it does solve the problem